source/auth/auth.rst

```diff
@@ -683,6 +683,239 @@ mechanism
+           Action=GetCallerIdentity&Version=2011-06-15
+|
+======================== ======================================================================================================
+Name                     Value       
+======================== ======================================================================================================
+HTTP Request Method      POST 
+URI                      /
+Content-Type*            application/x-www-form-urlencoded
+Content-Length*          43
+Host*                    Host field from Server First Message
+Region                   Derived from Host - see below
+X-Amz-Date*              See `Amazon Documentation <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+X-Amz-Security-Token*    Optional, See `Amazon Documentation <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+X-MongoDB-Server-Nonce*  Base64 string of server nonce
+X-MongoDB-GS2-CB-Flag*   ASCII lower-case character ‘n’ or ‘y’ or ‘p’
+X-MongoDB-Optional-Data* Optional, Optional data, base64 encoded representation of the optional object provided by the client
+Body                     Action=GetCallerIdentity&Version=2011-06-15\n
+======================== ======================================================================================================
+
+.. note::
+        ``*``, Denotes a header that MUST be included in SignedHeaders if present.
+
+.. note::
+        Region is not a header, but simply part of the authorization header. Region by default is ‘us-east-1’ since this is the 
+        implicit region for ‘sts.amazonaws.com’.  Drivers will need to derive the region to use from the endpoint. The region 
+        is the second piece of a FQDN name. While all official AWS STS endpoints start with “sts.”, there are non-AWS hosted 
+        endpoints and test endpoints that will not follow this rule.
+
+================= =========
+Host              Region
+================= =========
+sts.amazonaws.com us-east-1
+first.second      second
+first             us-east-1
+================= ========= 
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+	MAY be specified. The non-sensitive IAM access key.
+
+source
+	MUST be "$external". Defaults to ``$external``.
+
+password
+	MAY be specified. The sensitive IAM secret key.
+
+mechanism
+	MUST be "MONGODB-IAM"
+
+mechanism_properties
+	AWS_SESSION_TOKEN
+		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
+
+IAM Credentials
+   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers
+   MUST use this secret key to generate a signature using HMAC-SHA256.
```

Comment:
What are "IAM Credentials"? Is it supposed to be a field in `mechanism_properties`? How does a user provide the access key and secret key if not through the username+password?

Reply:
**What are "IAM Credentials"?** 
IAM Credentials are the access key and secret key. This is documented above, "MONGODB-IAM authenticates using AWS IAM credentials (an access key ID and a secret access key)".

**Is it supposed to be a field in mechanism_properties?**
No. The access key is the username and the secret key is password. It's documented under MongoCredential Properties as such, "username = MAY be specified. The non-sensitive IAM access key... password = MAY be specified. The sensitive IAM secret key. The "IAM Credentials" section is not a subsection of the "mechanism_properties" section.

**How does a user provide the access key and secret key if not through the username+password?**
Are you asking if there's another way to provide the access key and secret key besides setting the user/password in the URI? There is not. And if these are not provided we attempt to access temporary credentials by querying an AWS endpoint.

Reply:
I think it's confusing to put these sections under the "MongoCredential Properties" section. It makes is seem like "IAM Credentials" and "Temporary Credentials" are supposed to be fields of the MongoCredential type. Is that the case?

Can we add a section header above "IAM Credentials" to separate it from "MongoCredential Properties"?

Reply:
Done.